### ISSUE TITLE
Method overrides class in a incompatible manner (1.1.247)

### ISSUE THREAD

**Issue by dbrattli** [microsoft/pyright#3480]
**Describe the bug**

This might be a regression introduced by pyright 1.1.247. The error message does not make sense to me: 

`Type "_T2@map2" cannot be assigned to type "_T2@map2"`

<img width="972" alt="Screenshot 2022-05-19 at 07 43 38" src="https://user-images.githubusercontent.com/849479/169219241-eea06991-61b6-4c5f-9145-b02222fe8530.png">

**To Reproduce**
Steps to reproduce the behavior. See attached minimal repro:

**Expected behavior**

I would expect this code to pass type-checking, but would appreciate any feedback if I have misunderstood something here.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from __future__ import annotations
from abc import abstractmethod
from typing import (
    Callable,
    Generic,
    TypeVar,
)

_TSource = TypeVar("_TSource")
_TResult = TypeVar("_TResult")

_T2 = TypeVar("_T2")


class Option(Generic[_TSource]):
    @abstractmethod
    def map2(
        self, mapper: Callable[[_TSource, _T2], _TResult], other: Option[_T2]
    ) -> Option[_TResult]:
        raise NotImplementedError


class Some(Option[_TSource]):
    def map2(
        self, mapper: Callable[[_TSource, _T2], _TResult], other: Option[_T2]
    ) -> Option[_TResult]:
        return Nothing()


class Nothing(Option[_TSource]):
    def map2(
        self, mapper: Callable[[_TSource, _T2], _TResult], other: Option[_T2]
    ) -> Option[_TResult]:
        return Nothing()
```

**VS Code extension or command-line**

Pylance: v2022.5.2, Pyright 1.1.247

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression. I've fixed the problem and added a new unit test based on your sample to prevent future regressions in this code path.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.248, which I just published. It will also be included in the next release of pylance.
